service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/UrlsInResponseBodyRewriteFunction.java

```diff
@@ -32,7 +33,13 @@ public static String replaceUrl(String gpcConsumerUrl, String overrideGpcProvide
                 var gpcConsumerUrlPrefix = getUrlBase(exchange.getRequest().getURI());
                 LoggingUtil.debug(LOGGER, exchange, "The URL prefix for *this* GPC Consumer service is {}", gpcConsumerUrlPrefix);
 
-                URI proxyTargetUri = (URI) exchange.getAttributes().get(ServerWebExchangeUtils.GATEWAY_REQUEST_URL_ATTR);
+                URI proxyTargetUri = null;
+                if (exchange.getAttributes().containsKey(GATEWAY_REQUEST_URL_ATTR_BACKUP)) {
+                    proxyTargetUri = (URI) exchange.getAttributes().get(GATEWAY_REQUEST_URL_ATTR_BACKUP);
+                } else {
+                    proxyTargetUri = (URI) exchange.getAttributes().get(ServerWebExchangeUtils.GATEWAY_REQUEST_URL_ATTR);
```

Comment:
If GATEWAY_REQUEST_URL_ATTR is the more likely outcome, then switch this so it is checked first